components/supervisor/frontend/src/ide/heart-beat.ts

```diff
@@ -40,23 +42,27 @@ export function schedule(wsInfo: WorkspaceInfo, sessionId: string): void {
             console.error('Failed to send hearbeat:', err);
         }
     }
-    sendHeartBeat();
-    let unloadTimeout: any;
-    const beforeUnloadListener = () => {
-        unloadTimeout = setTimeout(() => {
-            // if unload was cancelled then resume heartbeating
-            sendHeartBeat();
-        }, 2000);
-        sendHeartBeat(true);
-    };
-    const unloadListener = () => {
-        if (unloadTimeout) {
-            clearTimeout(unloadTimeout);
+
+    const beaconWorkspacePageClose = async () => {
+        const instanceId = wsInfo.latestInstance!.id;
+        const data = { sessionId };
+        const blob = new Blob([JSON.stringify(data)], { type: "application/json" });
+        const url = serverUrl.withApi({ pathname: `/auth/workspacePageClose/${instanceId}` }).toString();
+        navigator.sendBeacon(url, blob);
+    }
+
+    let unloadListener = () => {
+        beaconWorkspacePageClose();
+    }
+
+    // TODO: Change version to correct one
+    if (!(await isSaasServerGreaterThan("main.4063"))) {
```

Comment:
Maybe if we do it earlier, i.e. then this file is loaded and store in some variable, something like
```
const serverVersion = (async () => {
   ....
}());
```

and isSaasServerGreaterThan awaits on it